app/controllers/rides_controller.rb

```diff
@@ -110,22 +99,43 @@ def edit
     @validation_pattern = VALIDATION_PATTERN
     I18n.locale = @ride.voter.locale
     @ride_zone = @ride.ride_zone
-    @pickup_at = @ride.pickup_in_time_zone
   end
 
   def update
     I18n.locale = @ride.voter.locale
+
+    # update the ride
     if @ride.update(ride_params)
-      render :success
-      @ride.conversation.send_from_staff(thanks_msg(@ride), Rails.configuration.twilio_timeout)
-      UserMailer.welcome_email_voter_ride(@ride.voter, @ride).deliver_later
+
+      # collect the params to use to update the user. move to method at some point
```

Comment:
This feels like a lot of logic for a controller. What do you think of something like
```rb
if params[:dispatcher] == 'true'
  @ride.update_voter(ride_params)
  redirect_to rides_dispatch_path(@ride.ride_zone.slug), notice: "Ride Updated!" and return
else
  @ride.update_and_notify_voter(ride_params)
  render :success
end
```
Or, if you can't get the asynchrony you want (delivering messages after the render), perhaps

```rb
@ride.update_voter(ride_params) do
  # happens after a successful update
  if params[:dispatcher] == 'true'
    redirect_to rides_dispatch_path(@ride.ride_zone.slug), notice: "Ride Updated!" and return
  else
    render :success
    @ride.notify_voter
  end
end
```

Either of these would also let you wrap the ride and voter updates in a transaction.

Reply:
(I think it's fine to leave a `# TODO` and come back to this after this PR)